File: Backend_dot_net_c_sharp_models_and_eums_to_convert_to_dart /Enumeration/Training/QualificationType.cs
using PeerageHRIS.Resources;
using PeerageHRIS.Resources.Models.Training;
using System.ComponentModel.DataAnnotations;

namespace PeerageHRIS.Enumeration.Training
{
    public enum QualificationType
    {
        [Display(Name = "Professional", ResourceType = typeof(TrainingResource))]
        Professional,
        [Display(Name = "Certification", ResourceType = typeof(TrainingResource))]
        Certification,
        [Display(Name = "Competency", ResourceType = typeof(TrainingResource))]
        Competency,
        [Display(Name = "Other", ResourceType = typeof(TrainingResource))]
        Other
    }
}

File: lib/core/enums/qualification_type.dart
import 'package:json_annotation/json_annotation.dart';

part 'qualification_type.g.dart';

@JsonEnum(alwaysCreate: true)
enum QualificationType {
  @JsonValue(0)
  professional,

  @JsonValue(1)
  certification,

  @JsonValue(2)
  competency,

  @JsonValue(3)
  other,
}
